crates/hotshot/src/traits/networking/memory_network.rs

```diff
@@ -302,22 +302,53 @@ impl<K: SignatureKey + 'static> ConnectedNetwork<K> for MemoryNetwork<K> {
         &self,
         message: Vec<u8>,
         recipients: Vec<K>,
-        broadcast_delay: BroadcastDelay,
+        _broadcast_delay: BroadcastDelay,
     ) -> Result<(), NetworkError> {
-        // Iterate over all topics, compare to recipients, and get the `Topic`
-        let topic = self
+        trace!(?message, "Broadcasting message to DA");
+        for node in self
             .inner
             .master_map
             .subscribed_map
+            .entry(Topic::Da)
+            .or_default()
             .iter()
-            .find(|v| v.value().iter().all(|(k, _)| recipients.contains(k)))
-            .map(|v| v.key().clone())
-            .ok_or(NetworkError::MessageSendError(
-                "no topic found for recipients".to_string(),
-            ))?;
-
-        self.broadcast_message(message, topic, broadcast_delay)
-            .await
+        {
+            if !recipients.contains(&node.0) {
+                tracing::error!("Skipping node because not in recipient list: {:?}", &node.0);
+                continue;
+            }
+            // TODO delay/drop etc here
+            let (key, node) = node;
+            trace!(?key, "Sending message to node");
+            if let Some(ref config) = &self.inner.reliability_config {
+                {
+                    let node2 = node.clone();
+                    let fut = config.chaos_send_msg(
```

Comment:
Is the `broadcast_message` function copied out to here? Curious - what's the difference in what we're doing between the two?

Reply:
Ah right, I forgot about that stipulation. SGTM, was just curious